generator/tests/test_other_library/run_other_library_tests.py

```diff
@@ -20,231 +20,289 @@
-    correct_file = '.\\test-code\\{2}\\{0}.{1}'.format(class_name, ext, global_variables.language)
-    temp_file = '.\\temp\\{2}\\{0}.{1}'.format(class_name, ext, global_variables.language)
-    return compare_files(correct_file, temp_file)
 
 def compare_binding_headers(class_name, binding, prefix):
-    correct_file = '.\\test-code\\{3}\\{0}\\{2}\\{1}.h'.format(binding, class_name, prefix, global_variables.language)
-    temp_file = '.\\temp\\{2}\\{0}\\{1}.h'.format(binding, class_name, global_variables.language)
-    return compare_files(correct_file, temp_file)
+    return compare_binding(class_name, binding, ".h", prefix)
 
 def compare_binding_impl(class_name, binding, prefix):
-    correct_file = '.\\test-code\\{3}\\{0}\\{2}\\{1}.cpp'.format(binding, class_name, prefix, global_variables.language)
-    temp_file = '.\\temp\\{2}\\{0}\\{1}.cpp'.format(binding, class_name, global_variables.language)
-    return compare_files(correct_file, temp_file)
+    return compare_binding(class_name, binding, ".cpp", prefix)
 
 def compare_binding_interface(class_name, binding, prefix):
-    correct_file = '.\\test-code\\{3}\\{0}\\{2}\\{1}.i'.format(binding, class_name, prefix, global_variables.language)
-    temp_file = '.\\temp\\{3}\\{0}\\{2}\\{1}.i'.format(binding, class_name, prefix, global_variables.language)
-    return compare_files(correct_file, temp_file)
+    return compare_binding(class_name, binding, ".i", prefix)
 
 def compare_binding_file(class_name, binding, prefix):
-    correct_file = '.\\test-code\\{3}\\{0}\\{2}\\{1}'.format(binding, class_name, prefix, global_variables.language)
-    temp_file = '.\\temp\\{3}\\{0}\\{2}\\{1}'.format(binding, class_name, prefix, global_variables.language)
-    return compare_files(correct_file, temp_file)
+    return compare_binding(class_name, binding, "", prefix)
+
+##########
 
-def compare_cmake_file(this_dir, prefix):
-    correct_file = '.\\test-code\\{2}\\cmake\\{1}\\{0}\\CMakeLists.txt'.format(this_dir, prefix, global_variables.language)
-    temp_file = '.\\temp\\{1}\\cmake\\{0}\\CMakeLists.txt'.format(this_dir, global_variables.language)
+def compare_cmake_file(this_dir):
+    """
+    Compare a temporary test CMakeLists.txt file with the reference one.
+
+    :param this_dir: lowest subdir containing the reference version of the file.
+    :return: 0 on success, or file not present; 1 on failure.
+    """
+    correct_file = os.path.normpath('./test-code/{0}/cmake/CMakeLists.txt'.format(gv.language))
+    temp_file = os.path.normpath('./temp/{1}/cmake/{0}/CMakeLists.txt'.format(this_dir, gv.language))
     return compare_files(correct_file, temp_file)
 
+
 #############################################################################
 # Specific test functions
 
+
 def run_test(name, num, class_name, test_case, list_of):
+    """
+    Run the C++ file tests.
+
+    :param name: name of test group e.g. 'test_sedml', 'combine-archive'.
+    :param num: list index TODO please clarify
```

Comment:
so we only generate a particular class not all of them

Reply:
docstring updated